feat(explore): add config for default time filter

Usiel opened the pull request:
### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

First change to complete https://github.com/apache/superset/issues/13125: We add a new configuration `DEFAULT_TIME_FILTER` to allow admins to set a global default time filter for any new explore charts. By default this is set to `None`, which replicates the previous behavior (`"No filter"`). To get a default filter of -1w the configuration can be set to `'Last week'` or any other value understood by Superset.

There is a minor change in behavior `UPDATE_FORM_DATA_BY_DATASOURCE` (when switching dataset in explore), the time filter is not reset anymore. This is similar to the behavior of other globally configured filters such as the row limit for explore, which also do not reset.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
New explore chart with `DEFAULT_TIME_FILTER = None`:
<img src="https://user-images.githubusercontent.com/1842905/196846013-317773cd-21e7-4ee3-889c-a16803874426.png" width="200">
New explore chart with `DEFAULT_TIME_FILTER = "Last week"`:
<img src="https://user-images.githubusercontent.com/1842905/196845999-d4838206-18d8-4ba3-bbdc-a9d93bed3957.png" width="200">

### TESTING INSTRUCTIONS

#### Test default behavior
1. Go to datasets and click on users (example dataset)
2. Add `updated` (time) as a dimension (helps to verify)

**Expected**: No time filter is applied when running the chart

#### Test with new configuration
1. Set `DEFAULT_TIME_FILTER = "2020-06-01T00:00:00 : now"` in `config.py`
2. Go to datasets and click on users (example dataset)
4. Add `updated` (time) as dimension (helps to verify)

**Expected**: Default time filter is applied

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [x] Has associated issue: https://github.com/apache/superset/issues/13125
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

Usiel committed:
feat: add config for default explore time filter

Related to issue https://github.com/apache/superset/issues/13125

First change to complete #13125: We add a new configuration `DEFAULT_TIME_FILTER` to allow admins to set a global deafult time filter for any explore charts. By default this is set to `None`, which replicates the previous behavior. To get a default filter of -1w the configuration can be set to `'Last week'`.

There is a minor change in behavior `UPDATE_FORM_DATA_BY_DATASOURCE`, the time filter is not reset anymore (the time dimension does still). This is similar to the behavior of other globally configured filters such as the row limit for explore.

Usiel commented:
Apologies for the forced pushes, was rebasing and missed something during my merge. I should be done now :)

zhaoyongjie commented on superset/config.py:
```diff
@@ -153,6 +153,9 @@ def _try_json_readsha(filepath: str, length: int) -> Optional[str]:
 FILTER_SELECT_ROW_LIMIT = 10000
+# default time filter in explore
+# values may be "Last day", "Last week", "<ISO date> : now", etc.
+DEFAULT_TIME_FILTER = None
```
one nite, we should keep this config as before, the `None` and the `No filter` will generate different time expressions. the `None` will generate ` : today`(col <= today()), and `No filter` will generate ` : `(empty filter).

Usiel replied:
Fixed 👍

Usiel committed:
Set explicit default for DEFAULT_TIME_FILTER

Using "No filter" string instead of None to avoid issues downstream if None is not properly handled.

zhaoyongjie reviewed (approved):
This config item is very useful in the production environment. LGTM, Thanks for the contribution!

Utsavjain4561 commented:
Hi, we are running superset 1.5.1rc1 right now and i guess in that there is no such support of `DEFAULT_TIME_FILTER` but i can see there a prop of `DEFAULT_TIME_RANGE` so is it possible if we hardcode its value to `Last week` and will it work for existing and new charts to have a default time range of `Last week` ?

zhaoyongjie commented:
> Hi, we are running superset 1.5.1rc1 right now and i guess in that there is no such support of `DEFAULT_TIME_FILTER` but i can see there a prop of `DEFAULT_TIME_RANGE` so is it possible if we hardcode its value to `Last week` and will it work for existing and new charts to have a default time range of `Last week` ?

Yes, it should be same as `DEFAULT_TIME_FILTER` in config.py.
